### PR TITLE
EA Notifications redesign

### PR DESCRIPTION
**Pull request by oetherington**
EA Forum notifications redesign ([relevant Figma](https://www.figma.com/design/qAHRGADVOGojqKpqbGMFFu/2024-Q2?node-id=1554-895&m=dev)).

Screenshot 2024-07-31 at 14 49 21

Screenshot 2024-07-31 at 14 49 33

Screenshot 2024-07-31 at 14 49 35

Screenshot 2024-07-31 at 14 49 25

Screenshot 2024-07-31 at 14 48 10

Screenshot 2024-07-31 at 14 46 57

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207924497791703) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/notifications/NotificationsMenuButton.tsx [hunk 1]
```diff
@@ -90,6 +92,11 @@ export const styles = (theme: ThemeType) => ({
     left: 1,
     top: 1,
   },
+  tooltip: {
```
Comment (oetherington) [inline packages/lesswrong/components/notifications/NotificationsMenuButton.tsx:95]: Moved from `MessagesMenuButton`

File: packages/lesswrong/components/notifications/NotificationsPopover.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+import { useCurrentUser } from "../common/withUser";
+import { styles as popoverStyles } from "../common/FriendlyHoverOver";
+import { useNotificationDisplays } from "./NotificationsPage/useNotificationDisplays";
+import { karmaSettingsLink } from "./NotificationsPage/NotificationsPageFeed";
+import type { NotificationDisplay } from "@/lib/notificationTypes";
+import type { KarmaChanges } from "@/lib/collections/users/karmaChangesGraphQL";
+import type { KarmaChangeUpdateFrequency } from "@/lib/collections/users/schema";
+import { AnalyticsContext } from "@/lib/analyticsEvents";
+
+const notificationsSettingsLink = "/account?highlightField=auto_subscribe_to_my_posts";
```
Comment (oetherington) [inline packages/lesswrong/components/notifications/NotificationsPopover.tsx:14]: I couldn't see any way to link to a _section_ on the settings page, so I just linked to the first setting in the section.

File: packages/lesswrong/components/notifications/NotificationsPopover.tsx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  }, [markAllAsRead]);
+
+  const closeMenu = useCallback(() => setIsOpen(false), []);
+
+  const closeAll = useCallback(() => {
+    closeMenu();
+    closePopover?.();
+  }, [closePopover, closeMenu]);
+
+  const notifs = useRef<NotificationDisplay[]>([]);
```
Comment (oetherington) [inline packages/lesswrong/components/notifications/NotificationsPopover.tsx:113]: Memoize notifications so they don't dissapear when loading more

File: packages/lesswrong/components/notifications/NotificationsPopover.tsx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+  if (
+    data?.NotificationDisplays?.results?.length &&
+    data.NotificationDisplays.results.length !== notifs.current.length
+  ) {
+    notifs.current = data.NotificationDisplays.results ?? [];
+  }
+
+  useEffect(() => {
+    if (!notificationsLoading && notifs.current.length <= defaultLimit) {
+      markAllAsRead?.();
```
Comment (oetherington) [inline packages/lesswrong/components/notifications/NotificationsPopover.tsx:123]: Only mark all as read once we've actually rendered the first batch

**Comment by oetherington** [comment]
Not sure if @agnesstenlund is available to review ATM, but FYI just in case

**Comment by agnesstenlund** [comment]
This looks really great 🥲 

I made some minor padding updates (I can see you built them exactly as the mockup so this is just me changing my mind after seeing it live).

One thing I didn't change but would like to:
- [ ] Would be great if the icon and text look top-aligned (I know they are, but it looks like the icon is 2px further up)

**Comment by agnesstenlund** [comment]
- [ ] The unread dot looks a little far from the number
- [ ] It also looked smaller in production than expected–I'd bump it up to 10px actually

**Review by s-cheng — APPROVED** [review]
Looks really nice, thanks Ollie! 🙌

File: packages/lesswrong/components/notifications/NotificationsPopover.tsx [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+            open={isOpen}
+            anchorEl={anchorEl.current}
+            placement="bottom-end"
+          >
+            <LWClickAwayListener onClickAway={closeMenu}>
+              <DropdownMenu>
+                <DropdownItem
+                  title="Mark all as read"
+                  onClick={markAllAsRead}
+                />
```
[the remark this answers is not in the document]
Comment (oetherington) [reply packages/lesswrong/components/notifications/NotificationsPopover.tsx:161]: Hmm, good point. I'm going to leave it for now as I think it's a decent fail-safe incase there's any issues with the logic for clearing when the popover opens (I know we've had issues with stuff like that in the past), but we can probably remove this in the future if no bugs present themselves.

**Pull request merged**